### ISSUE TITLE
Support AWS hosted Elasticsearch service

### ISSUE THREAD

**Issue by devinsba** [openzipkin/zipkin#1253]
In order to use the AWS hosted Elasticsearch service we will need to support the REST api instead of the native transport API.

I'm wondering if this should be a separate module or a flag in the ElasticSearch storage builder if I were to create a PR for this.

**Comment by codefromthecrypt** [comment]
Unless the libraries change or the code ends up way different, I would make
it a flag or similar. I presume there's feature parity across the two?

**Comment by devinsba** [comment]
I don't know about feature parity. I do think it is a different client
library to talk over REST. I only gave it a quick glance at the end of the
day though since I had some time to kill at work. I'll follow up when I get
my computer back in the morning

**Comment by codefromthecrypt** [comment]
In the mean time, pinged on twitter about this as this is a feature requested for them many times. It would be nice to understand if this is a backlog item on the product or not.

https://twitter.com/adrianfcole/status/768616436948271104

I noticed that maggieaws is the one doing product launch announcements on the forum:
https://forums.aws.amazon.com/ann.jspa?annID=3946

This might be the correct maggie, or maybe just there are multiple AWS product managers named maggie
https://www.linkedin.com/in/yuanzhuang

If someone knows the product team, please reach out and see if this is on the roadmap as rewriting would be fruitless if it is.

**Comment by anuraaga** [comment]
There would never be a feature difference between the Transport client and REST since non-Java clients have to use the REST client ;)

The Transport client is a lot more efficient thanks to somewhat more compact serialization and multiplexing over the wire. ES unfortunately doesn't support HTTP2 on the rest port so the lack of multiplexing can matter at high loads.

I think it's possible to convert a request object that we construct already directly into a json object that can be POSTed, so supporting both may not be so bad. Though I generally don't recommend AWS managed ES since going by history, it'll be a year behind the times again when 5.0 comes out...

**Comment by codefromthecrypt** [comment]
PS elastic just mentioned that their cloud (which runs on AWS) supports transport client. So, that's one way to get collocated data in AWS. https://twitter.com/elastic/status/768621013063995392

**Comment by codefromthecrypt** [comment]
there's a new impl in town, via @sethp-jive https://github.com/openzipkin/zipkin/pull/1302

**Comment by codefromthecrypt** [comment]
Will be released in zipkin 1.12

``` bash
# make sure your cli credentials are setup as zipkin will read them
$ aws es describe-elasticsearch-domain --domain-name mydomain|jq .DomainStatus.Endpoint
"search-mydomain-2rlih66ibw43ftlk4342ceeewu.ap-southeast-1.es.amazonaws.com"
$ STORAGE_TYPE=elasticsearch ES_HOSTS=https://search-mydomain-2rlih66ibw43ftlk4342ceeewu.ap-southeast-1.es.amazonaws.com java -jar zipkin.jar
# or if you don't know the url already
$ STORAGE_TYPE=elasticsearch ES_AWS_DOMAIN=mycluster ES_AWS_REGION=ap-southeast-1 java -jar zipkin.jar
```
